src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/temporary/TemporaryEdge.java

```diff
@@ -0,0 +1,51 @@
+package org.openstreetmap.atlas.geography.atlas.raw.slicing.temporary;
+
+import java.util.Map;
+
+import org.openstreetmap.atlas.geography.PolyLine;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+
+/**
+ * The {@link TemporaryEdge} object, keeps track of the bare minimum information required to create
+ * an Atlas {@link Edge}. It is meant to be as light-weight as possible, keeping track of only the
+ * identifier, polyLine, tags and whether there is an associated reverse Edge.
+ *
+ * @author mgostintsev
+ */
+public class TemporaryEdge extends TemporaryEntity
+{
+    private static final long serialVersionUID = 3867946360797866502L;
+
+    private final PolyLine polyLine;
+    private final boolean hasReverse;
+
+    public TemporaryEdge(final long identifier, final PolyLine polyLine,
+            final Map<String, String> tags, final boolean hasReverse)
+    {
+        super(identifier, tags);
+        this.polyLine = polyLine;
+        this.hasReverse = hasReverse;
+    }
+
+    public PolyLine getPolyLine()
+    {
+        return this.polyLine;
+    }
+
+    public long getReversedIdentifier()
+    {
+        return -1 * Math.abs(getIdentifier());
```

Comment:
If the identifier passed in is negative to start with, we could end up always returning the negative identifier.  Should this be return -1 * getIdentifier()?

Reply:
We will never start off with a negative identifier, will add a comment to clarify.

Reply:
Ah, then maybe we don't need to wrap it in the Math.abs() as well, that was what made me think of the negative number case :)

Reply:
Yep! Ended up removing that call.